src/tools/floptool.cpp

```diff
@@ -26,16 +27,21 @@
 
 static formats_table formats;
 
-static void display_usage()
+static void display_usage(const char *first_argument)
 {
+	std::string exe_name(core_filename_extract_base(first_argument));
+
 	fprintf(stderr, "Usage: \n");
-	fprintf(stderr, "       floptool.exe identify <inputfile> [<inputfile> ...]                                 -- Identify an image format\n");
-	fprintf(stderr, "       floptool.exe flopconvert [input_format|auto] output_format <inputfile> <outputfile> -- Convert a floppy image\n");
-	fprintf(stderr, "       floptool.exe flopcreate output_format filesystem <outputfile>                       -- Create a preformatted floppy image\n");
-	fprintf(stderr, "       floptool.exe flopdir input_format filesystem <image>                                -- List the contents of a floppy image\n");
-	fprintf(stderr, "       floptool.exe flopread input_format filesystem <image> <path> <outputfile>           -- Extract a file from a floppy image\n");
-	fprintf(stderr, "       floptool.exe flopwrite input_format filesystem <image> <inputfile> <path>           -- Write a file into a floppy image\n");
-	fprintf(stderr, "       floptool.exe version                                                                -- Display the current version of floptool\n");
+	fprintf(stderr, "       %s identify <inputfile> [<inputfile> ...]                                 -- Identify an image format\n", exe_name.c_str());
+	fprintf(stderr, "       %s flopconvert [input_format|auto] output_format <inputfile> <outputfile> -- Convert a floppy image\n", exe_name.c_str());
+	fprintf(stderr, "       %s flopcreate output_format filesystem <outputfile>                       -- Create a preformatted floppy image\n", exe_name.c_str());
+	fprintf(stderr, "       %s flopdir input_format filesystem <image>                                -- List the contents of a floppy image\n", exe_name.c_str());
+	fprintf(stderr, "       %s flopread input_format filesystem <image> <path> <outputfile>           -- Extract a file from a floppy image\n", exe_name.c_str());
+	fprintf(stderr, "       %s flopwrite input_format filesystem <image> <inputfile> <path>           -- Write a file into a floppy image\n", exe_name.c_str());
+	fprintf(stderr, "       %s hddir filesystem <image>                                               -- List the contents of a floppy image\n", exe_name.c_str());
+	fprintf(stderr, "       %s hdread filesystem <image> <path> <outputfile>                          -- Extract a file from a floppy image\n", exe_name.c_str());
+	fprintf(stderr, "       %s hdwrite filesystem <image> <inputfile> <path>                          -- Write a file into a floppy image\n", exe_name.c_str());
```

Comment:
There seems to be some faulty copy-and-paste here, as I would guess none of these three are "floppy image" commands.